subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/ValueSnapshotter.java

```diff
@@ -129,6 +142,17 @@ private SerializedValueSnapshot serialize(Object value) {
         return new SerializedValueSnapshot(classLoaderHasher.getClassLoaderHash(value.getClass().getClassLoader()), outputStream.toByteArray());
     }
 
+    private ValueSnapshot wrap(Object value, ValueSnapshot possible) {
+        if (possible instanceof EnumValueSnapshot) {
```

Comment:
We don't want to retain strong references to the enum values or the value's class when we only need a snapshot (rather than an isolatable), for example when we're snapshotting a task's properties. I would add another public method whose responsibility is to create an `Isolatable` and leave the behaviour of `snapshot()` as it was, and use this new method for artifact transforms registration.